### PR TITLE
#113: move-assign in eastl::remove()/remove_if().

### PR DESCRIPTION
**Pull request by Vuhdo**

### CODE REVIEW

File: include/EASTL/algorithm.h [hunk 1]
```diff
@@ -2670,10 +2670,16 @@ namespace eastl
 	remove(ForwardIterator first, ForwardIterator last, const T& value)
 	{
 		first = eastl::find(first, last, value);
-		if(first != last)
+		if(first != last) 
 		{
-			ForwardIterator i(first);
-			return eastl::remove_copy(++i, last, first, value);
+			for(ForwardIterator i = first; ++i != last;)
```
Comment (rparolin) [inline include/EASTL/algorithm.h]: Please give this pattern a name like the code previously did.  ie.  "eastl::remove_copy"...  Perhaps eastl::remove_move?  I'm open to other/better names. :)

File: include/EASTL/algorithm.h [hunk 2]
```diff
@@ -2707,10 +2713,16 @@ namespace eastl
 	remove_if(ForwardIterator first, ForwardIterator last, Predicate predicate)
 	{
 		first = eastl::find_if(first, last, predicate);
-		if(first != last)
+		if(first != last) 
 		{
-			ForwardIterator i(first);
-			return eastl::remove_copy_if<ForwardIterator, ForwardIterator, Predicate>(++i, last, first, predicate);
+			for(ForwardIterator i = first; ++i != last;)
```
Comment (rparolin) [inline include/EASTL/algorithm.h]: Ditto above.  eastl:remove_move_if?

**Comment by rparolin** [comment]
Not sure why Appveyor is failing here as there is no link and I can't seem to find the bad build on the Appveyor site directly.  Wierd.

**Comment by rparolin** [comment]
Just following up, to see if you had any info on why Appveyor failed here?

**Comment by theryee** [comment]
Couldn't figure it out. I had to take a break from this PR but I'll have some time later this week to follow up.

**Comment by rparolin** [comment]
No worries.  Thanks for looking into this.

**Comment by rparolin** [comment]
Gonna close this.  Feel free to reopen / resubmit when you have more time.  Thanks.

**Pull request closed without merging**